spec/integration/parser/compiler_spec.rb

```diff
@@ -898,6 +898,24 @@ class { 'foo':  }
         end.to raise_error(/Class\[Foo\]: parameter 'x' expects an Integer value, got Undef/)
       end
 
+      it 'denies a regexp (rich data) argument given to class String parameter (even if later encoding of it is a string)' do
+        expect do
+          catalog = compile_to_catalog(<<-MANIFEST)
+            class foo(String $x) { }
+            class { 'foo':  x => /I am a regexp and I don't want to be a String/}
```

Comment:
Perhaps you could use a regexp containing escaped forward slashes to ensure that escaping is consistent.

Reply:
We already have such a test elsewhere - it ensures that a regexp is formatted as expected.

Reply:
For the particular case when the new unknown_to_string method is used?